src/policy/fees.cpp

```diff
@@ -690,6 +690,20 @@ void CBlockPolicyEstimator::processBlock(const std::vector<RemovedMempoolTransac
     shortStats->UpdateMovingAverages();
     longStats->UpdateMovingAverages();
 
+    std::set<Txid> seen_transactions;
+    for (const auto& tx : txs_removed_for_block) {
+        seen_transactions.insert(tx.info.m_tx->GetHash());
+        for (const auto& input : tx.info.m_tx->vin) {
+            const Txid& parentId = input.prevout.hash;
+            if (seen_transactions.count(parentId)) {
+                // Ignore any transaction with a child in the received block
+                // It may be fee bumped by the child.
+                _removeTx(parentId, /*inblock=*/true);
+                seen_transactions.erase(parentId);
+            }
+        }
+    }
```

Comment:
Nit: This code block can be extracted in a separate function because after reading the code above that's mostly delegation consecutive function calls, it'd be nice to have something similar here as well - `removeParentTxs`. One more point being that `seen_transactions` variable is internal to this logic and doesn't need to be exposed to `processBlock`.

Reply:
Fixed